api/clients/v2/coretypes/eigenda_cert.go

```diff
@@ -4,34 +4,206 @@ import (
 	"fmt"
 
 	disperser "github.com/Layr-Labs/eigenda/api/grpc/disperser/v2"
-	contractEigenDACertVerifier "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDACertVerifierV2"
+	v2_cert_verifier "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDACertVerifierV2"
+	cert_types_binding "github.com/Layr-Labs/eigenda/contracts/bindings/IEigenDACertTypeBindings"
 	v2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/rlp"
 )
 
-// EigenDACert contains all data necessary to retrieve and validate a blob
-//
-// This struct represents the composition of a eigenDA blob certificate, as it would exist in a rollup inbox.
-type EigenDACert struct {
-	BlobInclusionInfo           contractEigenDACertVerifier.EigenDATypesV2BlobInclusionInfo
-	BatchHeader                 contractEigenDACertVerifier.EigenDATypesV2BatchHeaderV2
-	NonSignerStakesAndSignature contractEigenDACertVerifier.EigenDATypesV1NonSignerStakesAndSignature
+var (
+	v3CertTypeEncodeArgs abi.Arguments
+)
+
+func init() {
+	// load the ABI and parse the dummy interface methods used to encode the cert
+	// NOTE: the only other way would be defining the certificate using go-ethereum's abi
+	// low level types which would require much boiler plate
+	certTypesBinding, err := cert_types_binding.ContractIEigenDACertTypeBindingsMetaData.GetAbi()
+	if err != nil {
+		panic(err)
+	}
+
+	v3CertTypeEncodeMethod, ok := certTypesBinding.Methods["dummyFnCertV3"]
+	if !ok {
+		panic("dummyFnCertV3 not found in IEigenDACertTypes ABI")
+	}
+
+	v3CertTypeEncodeArgs = v3CertTypeEncodeMethod.Inputs
+
+}
+
+// CertificateVersion denotes the version of the EigenDA certificate
+// and is interpreted from querying the EigenDACertVerifier contract's
+// CertVersion() view function
+type CertificateVersion = byte
+
+const (
+	// starting at two since we never formally defined a V1 cert in the core code
+	VersionTwoCert   = 0x2
+	VersionThreeCert = 0x3
+)
+
+type EigenDACert interface {
+	BlobVersion() v2.BlobVersion
+	ReferenceBlockNumber() uint64
+	QuorumNumbers() []byte
+	ComputeBlobKey() (*v2.BlobKey, error)
+	Serialize() ([]byte, error)
+	RelayKeys() []v2.RelayKey
+	Commitments() (*encoding.BlobCommitments, error)
```

Comment:
Feel like we should be returning bindings types and not backend types (encoding.BlobCommitments, v2.BlobKey, etc) here since the EigenDACert types themselves only contain the binding types, so why convert them back to backend types?

Reply:
But integration shouldn't care about core backend types. We should be agnostic to those types, to allow backend to do refactors/rewrites without affecting us. Basic encapsulation. Also the cert's main purpose is to be verified onchain by the CertVerifier contract, so it should contain those types.

Reply:
how would that work with additional client package dependencies (e.g, distributed retrieval) that expect core types to be extractable from the certificate?

Reply:
> how would that work with additional client package dependencies (e.g, distributed retrieval)

The preexisting cert structure is already implemented with bindings, so there wouldn't be any change to the clients in that regard.

Reply:
the preexisting cert structure uses type bindings for `EigenDACertVerifierV2` so there would need to be translation

Reply:
I see- so is the main problem from your perspective that, in the short term as we transition to this new cert type, we would need translation between the different binding types?

Or do you think it is fundamentally incorrect to use binding types within the cert interface here?

Reply:
> Or do you think it is fundamentally incorrect to use binding types within the cert interface here?

Given we already have structured core types that are expected by the other clients I don't see any merit in using bindings directly.